subsys/bluetooth/controller/hci_cmd_put.c

```diff
@@ -0,0 +1,851 @@
+	cmds->hci_le_set_address_resolution_enable = 1;
+	cmds->hci_le_set_resolvable_private_address_timeout = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+	cmds->hci_le_read_maximum_data_length = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_PHY)
+	cmds->hci_le_read_phy = 1;
+	cmds->hci_le_set_default_phy = 1;
+	cmds->hci_le_set_phy = 1;
+#endif
+
+#if defined(BT_CTLR_DTM)
+	cmds->hci_le_receiver_test_v2 = 1;
+	cmds->hci_le_transmitter_test_v2 = 1;
+#endif
+
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_BROADCASTER)
+	cmds->hci_le_set_advertising_set_random_address = 1;
+	cmds->hci_le_set_extended_advertising_parameters = 1;
+	cmds->hci_le_set_extended_advertising_data = 1;
+	cmds->hci_le_set_extended_scan_response_data = 1;
+	cmds->hci_le_set_extended_advertising_enable = 1;
+	cmds->hci_le_read_maximum_advertising_data_length = 1;
+	cmds->hci_le_read_number_of_supported_advertising_sets = 1;
+	cmds->hci_le_remove_advertising_set = 1;
+	cmds->hci_le_clear_advertising_sets = 1;
+#endif
+
+#if defined(CONFIG_BT_OBSERVER)
+	cmds->hci_le_set_extended_scan_parameters = 1;
+	cmds->hci_le_set_extended_scan_enable = 1;
+#endif
+#if defined(CONFIG_BT_CENTRAL)
+	cmds->hci_le_extended_create_connection = 1;
+#endif
+#endif
+	cmds->hci_le_read_transmit_power = 1;
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+	cmds->hci_le_set_privacy_mode = 1;
+#endif
+}
+
+static void le_read_supported_states(uint64_t *buf)
+{
+	uint64_t states = 0U;
+
+#define ST_ADV (BIT64(0)  | BIT64(1)  | BIT64(8)  | BIT64(9)  | BIT64(12) | \
+		BIT64(13) | BIT64(16) | BIT64(17) | BIT64(18) | BIT64(19) | \
+		BIT64(20) | BIT64(21))
+
+#define ST_SCA (BIT64(4)  | BIT64(5)  | BIT64(8)  | BIT64(9)  | BIT64(10) | \
+		BIT64(11) | BIT64(12) | BIT64(13) | BIT64(14) | BIT64(15) | \
+		BIT64(22) | BIT64(23) | BIT64(24) | BIT64(25) | BIT64(26) | \
+		BIT64(27) | BIT64(30) | BIT64(31))
+
+#define ST_SLA (BIT64(2)  | BIT64(3)  | BIT64(7)  | BIT64(10) | BIT64(11) | \
+		BIT64(14) | BIT64(15) | BIT64(20) | BIT64(21) | BIT64(26) | \
+		BIT64(27) | BIT64(29) | BIT64(30) | BIT64(31) | BIT64(32) | \
+		BIT64(33) | BIT64(34) | BIT64(35) | BIT64(36) | BIT64(37) | \
+		BIT64(38) | BIT64(39) | BIT64(40) | BIT64(41))
+
+#define ST_MAS (BIT64(6)  | BIT64(16) | BIT64(17) | BIT64(18) | BIT64(19) | \
+		BIT64(22) | BIT64(23) | BIT64(24) | BIT64(25) | BIT64(28) | \
+		BIT64(32) | BIT64(33) | BIT64(34) | BIT64(35) | BIT64(36) | \
+		BIT64(37) | BIT64(41))
+
+#if defined(CONFIG_BT_BROADCASTER)
+	states |= ST_ADV;
+#else
+	states &= ~ST_ADV;
+#endif
+#if defined(CONFIG_BT_OBSERVER)
+	states |= ST_SCA;
+#else
+	states &= ~ST_SCA;
+#endif
+#if defined(CONFIG_BT_PERIPHERAL)
+	states |= ST_SLA;
+#else
+	states &= ~ST_SLA;
+#endif
+#if defined(CONFIG_BT_CENTRAL)
+	states |= ST_MAS;
+#else
+	states &= ~ST_MAS;
+#endif
+	/* All states and combinations supported except:
+	 * Initiating State + Passive Scanning
+	 * Initiating State + Active Scanning
+	 */
+	states &= ~(BIT64(22) | BIT64(23));
```

Comment:
How about removing them from the definitions above? They can be added back when they are supported.

Reply:
I've copied over the implementation from the zephyr LL. I would prefer to keep it the same

Reply:
ok, fine by me